packages/node-core/src/indexer/blockDispatcher/base-block-dispatcher.ts

```diff
@@ -169,22 +170,31 @@ export abstract class BaseBlockDispatcher<Q extends IQueue, DS, B> implements IB
   // Is called directly after a block is processed
   @mainThreadOnly()
   protected async postProcessBlock(height: number, processBlockResponse: ProcessBlockResponse): Promise<void> {
-    const {blockHash, dynamicDsCreated, reindexBlockHeight} = processBlockResponse;
+    const {blockHash, dynamicDsCreated, reindexBlockHeight: processReindexBlockHeight} = processBlockResponse;
+    // Rewind height received from admin api have higher priority than processed reindexBlockHeight
+    const reindexBlockHeight = this._pendingRewindHeight ?? processReindexBlockHeight;
     this.monitorService?.write(`Finished block ${height}`);
     if (reindexBlockHeight !== null && reindexBlockHeight !== undefined) {
-      if (this.nodeConfig.proofOfIndex) {
-        await this.poiSyncService.stopSync();
-        this.poiSyncService.clear();
-        this.monitorService?.write(`poiSyncService stopped, cache cleared`);
+      try {
+        if (this.nodeConfig.proofOfIndex) {
+          await this.poiSyncService.stopSync();
+          this.poiSyncService.clear();
+          this.monitorService?.write(`poiSyncService stopped, cache cleared`);
+        }
+        this.monitorService?.createBlockFork(reindexBlockHeight);
```

Comment:
Do we need a way to distinguish between a fork rewind and a user rewind?

Reply:
From a consequence perspective, fork and manual rewind by the user have the same effect. Therefore, I think they can both be marked as fork.
When user rewind we also added an additional message,  https://github.com/subquery/subql/pull/2415/files#diff-cae421fb50a0ad66cd7a4b1cb3e21eccb811a96a765ce08761c83ac3fd31156fR244 then it is not difficult for us to trace it from log